src/riscv/lib/src/interpreter/load_store.rs

```diff
@@ -82,9 +83,50 @@ pub fn run_load<V: StoreLoadInt, I: ICB>(
     })
 }
 
+/// Stores a double-precision floating point value `val(rs2)` to the address
+/// starting at `val(rs1) + imm`.
+pub fn run_f64_store<I: ICB>(
+    icb: &mut I,
+    imm: i64,
+    rs1: XRegister,
+    rs2: FRegister,
+) -> I::IResult<()> {
+    let rs1val = icb.xregister_read(rs1);
+    let imm = icb.xvalue_of_imm(imm);
+
+    let address = rs1val.add(imm, icb);
+
+    let fval = icb.fregister_read(rs2);
+    let xval = icb.fvalue_to_xvalue(fval);
+
+    icb.main_memory_store::<u64>(address, xval)
+}
+
+/// Loads a double-precision floating point value from the address
+/// starting at `val(rs1) + imm` and writes it to `rd`.
+pub fn run_f64_load<I: ICB>(
+    icb: &mut I,
+    imm: i64,
+    rs1: XRegister,
+    rd: FRegister,
+) -> I::IResult<()> {
+    let rs1val = icb.xregister_read(rs1);
+    let imm = icb.xvalue_of_imm(imm);
+
+    let address = rs1val.add(imm, icb);
+
+    let res = icb.main_memory_load::<u64>(address);
```

Comment:
Extend stores and loads to floats, please. Reading as an integer is not the same. They have different endianness criteria, see [this impls](https://github.com/tezos/riscv-pvm/blob/f92a1f4e257ed58fd7ff0a1a3990792e4e2a76de/src/riscv/lib/src/machine_state/registers.rs#L583-L599) versus [this impl](https://github.com/tezos/riscv-pvm/blob/f92a1f4e257ed58fd7ff0a1a3990792e4e2a76de/src/riscv/lib/src/state_backend/elems.rs#L65-L84).